### PR TITLE
Update gt4py

### PR DESCRIPTION
**Pull request by jdahm**
## Purpose

Update gt4py to [5050308](https://github.com/GridTools/gt4py/commit/50503087eae255be4bb2cc5ede0da27f3696e424) from 2022.05.20.

## Changes

The gt4py backends are renamed to drop the `gtc:` prefix.

### CODE REVIEW

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
You should update `is_gtc_backend` in `dsl/pace/dsl/stencil.py` and `pace-util/pace/util/global_config.py`, possibly consolidating these two. Also `pace-util/pace/util/testing/gt4py_cupy.py`, `pace-util/tests/test_halo_data_transformer.py`

We have some possibly-unused data referring to the "gtcuda" backend, e.g. docker/Makefile.image_names, fv3core/Makefile, fv3core/docker/Makefile.image_names and fv3core/tests/savepoint/translate/overrides/standard.yaml. Basically `git grep gtcuda .` and `git grep gtx86 .`. Not sure if you should update these here or if they're a separate issue, if they're a separate issue let's do it in a separate PR (even if you're the one doing it).

File: dsl/pace/dsl/stencil.py [hunk 1]
```diff
@@ -103,17 +103,16 @@ def _get_backend_opts(
         return backend_opts
 
     def stencil_kwargs(self, skip_passes: Iterable[str] = ()):
+        # NOTE (jdahm): Temporary replace call until Jenkins is updated
         kwargs = {
-            "backend": self.backend,
+            "backend": self.backend.replace("gtc:", ""),
```
Comment (mcgibbon) [inline dsl/pace/dsl/stencil.py:108]: Remember to remove this later.

File: dsl/pace/dsl/stencil.py [hunk 2]
```diff
@@ -103,17 +103,16 @@ def _get_backend_opts(
...
             "rebuild": self.rebuild,
             **self.backend_opts,
         }
         if not self.is_gpu_backend:
             kwargs.pop("device_sync", None)
         # Note: this assure the backward compatibility between v36 and v37
-        if self.backend.startswith("gtc") and (
-            "skip_passes" in kwargs or skip_passes is not None
-        ):
+        if "skip_passes" in kwargs or skip_passes is not None:
```
Comment (mcgibbon) [inline dsl/pace/dsl/stencil.py:115]: I know your refactor hasn't changed this code, but is this code wrong? skip_passes can never be `None`.

File: pace-util/pace/util/initialization/allocator.py [hunk 3]
```diff
@@ -24,6 +24,7 @@ def __init__(self, backend: str):
         Args:
             backend: gt4py backend
         """
+        backend = backend.replace("gtc:", "")
```
Comment (mcgibbon) [inline pace-util/pace/util/initialization/allocator.py:27]: Possibly also a TODO/cleanup needed here.

**Review by mcgibbon — COMMENTED** [review]
See previous comment for some required changes.

File: dsl/pace/dsl/stencil.py (hunk 2 above)
Comment (jdahm) [reply dsl/pace/dsl/stencil.py:115]: Removed. Nice catch!

File: dsl/pace/dsl/gt4py_utils.py [hunk 4]
```diff
@@ -261,7 +264,8 @@ def make_storage_from_shape(
             mask = (False, False, True)  # Assume 1D is a k-field
         else:
             mask = (n_dims * (True,)) + ((3 - n_dims) * (False,))
-
+    # NOTE (jdahm): Temporary until Jenkins is updated
+    backend = backend.replace("gtc:", "")
```
Comment (mcgibbon) [inline dsl/pace/dsl/gt4py_utils.py:268]: Maybe make a Jira task so you can't forget.

File: dsl/pace/dsl/stencil.py [hunk 5]
```diff
@@ -103,29 +99,27 @@ def _get_backend_opts(
...
     @property
     def is_gpu_backend(self) -> bool:
-        return self.backend.endswith("cuda") or self.backend.endswith("gpu")
-
-    @property
-    def is_gtc_backend(self) -> bool:
-        return self.backend.startswith("gtc")
+        try:
+            return gt4py.backend.from_name(self.backend).storage_info["device"] == "gpu"
+        except Exception:
```
Comment (mcgibbon) [inline dsl/pace/dsl/stencil.py:120]: can you catch any more specific exception than this?

File: fv3core/tests/main/dsl/test_stencil_wrapper.py [hunk 6]
```diff
@@ -262,23 +260,20 @@ def test_backend_options(
...
+            "backend": "cuda",
             "rebuild": True,
             "device_sync": False,
             "format_source": False,
-            "oir_pipeline": gtc.passes.oir_pipeline.DefaultPipeline(),
             "verbose": False,
         },
     }
 
-    set_backend(backend)
```
Comment (mcgibbon) [inline fv3core/tests/main/dsl/test_stencil_wrapper.py:281]: Thanks for removing this!

File: pace-util/pace/util/global_config.py [hunk 7]
```diff
@@ -11,8 +11,6 @@ def getenv_bool(name: str, default: str) -> bool:
 def set_backend(new_backend: str):
     global _BACKEND
     _BACKEND = new_backend
-    for function in (is_gpu_backend, is_gtc_backend):
```
Comment (mcgibbon) [inline pace-util/pace/util/global_config.py:14]: Can we delete this file entirely now (possibly in a later PR)?

**Review by mcgibbon — APPROVED** [review]
If you can catch a more specific exception please do, otherwise LGTM!

File: pace-util/pace/util/global_config.py (hunk 7 above)
Comment (jdahm) [reply pace-util/pace/util/global_config.py:14]: Yes, in a follow up this afternoon!

**Comment by jdahm** [comment]
The 56 rank tests caught a problem that was caused by a change in gt4py. Discussing that [here](https://github.com/GridTools/gt4py/issues/755#issuecomment-1130586373).

**Comment by jdahm** [comment]
launch jenkins

**Pull request merged**